tests/CMakeLists.txt

```diff
@@ -15,12 +15,18 @@ target_sources(frozen.tests PRIVATE
   ${CMAKE_CURRENT_LIST_DIR}/test_unordered_set.cpp
   ${CMAKE_CURRENT_LIST_DIR}/test_unordered_str_set.cpp)
 
-add_compile_options(frozen.tests PUBLIC
-  $<$<CXX_COMPILER_ID:MSVC>:/w4>
-  $<$<OR:$<CXX_COMPILER_ID:GNU>,
-         $<CXX_COMPILER_ID:GNU>>:-Wextra -W -Werror -Wshadow>
-  $<$<CXX_COMPILER_ID:Intel>:
-    $<$<PLATFORM_ID:Windows>:/Wall>
-    $<$<NOT:$<PLATFORM_ID:Windows>>:-Wall>>
-  )
+string( CONCAT generator
+  # msvc gives invalid integral overflow warning for unsigned type
+  "$<$<CXX_COMPILER_ID:MSVC>:/W3;/WX;/wd4307>"
```

Comment:
I would suggest adding a corresponding generator to the frozen target, e.g.

```cmake
target_compile_option(frozen INTERFACE $<$<CXX_COMPILER_ID:MSVC>:/wd4307;>>)
```

If you're confident that the overflow warning is erroneous, I'll jump through the hoops to file a bug with the visual studios team.

Reply:
OK for filling a bug, but not ok for adding this flag to the compile options, as it has a global effect : I don't wan't to force that to the users. We should document this behavior in the README though.

Reply:
Seems reasonable